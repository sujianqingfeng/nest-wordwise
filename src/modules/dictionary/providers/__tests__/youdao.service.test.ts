import { HttpModule } from '@nestjs/axios'
import { Test, TestingModule } from '@nestjs/testing'
import { YouDaoDictionaryService } from '../youdao.service'

describe('YouDaoDictionaryService', () => {
  let service: YouDaoDictionaryService

  beforeEach(async () => {
    const module: TestingModule = await Test.createTestingModule({
      providers: [YouDaoDictionaryService],
      imports: [HttpModule]
    }).compile()

    service = module.get<YouDaoDictionaryService>(YouDaoDictionaryService)
  })

  it('should be defined', () => {
    expect(service).toBeDefined()
  })

  it('should query', async () => {
    const result = await service.query('hello')
    expect(result).toMatchSnapshot()
  })
})
