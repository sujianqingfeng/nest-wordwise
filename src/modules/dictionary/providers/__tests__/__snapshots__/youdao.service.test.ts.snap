// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YouDaoDictionaryService should query 1`] = `
{
  "forms": [
    {
      "name": "复数",
      "value": "hellos",
    },
    {
      "name": "第三人称单数",
      "value": "helloes",
    },
    {
      "name": "现在分词",
      "value": "helloing",
    },
    {
      "name": "过去式",
      "value": "helloed",
    },
    {
      "name": "过去分词",
      "value": "helloed",
    },
  ],
  "translates": [
    {
      "position": "int.",
      "translate": "喂，你好（用于问候或打招呼）；喂，你好（打电话时的招呼语）；喂，你好（引起别人注意的招呼语）；<非正式>喂，嘿 (认为别人说了蠢话或分心)；<英，旧>嘿（表示惊讶）",
    },
    {
      "position": "n.",
      "translate": "招呼，问候；（Hello）（法、印、美、俄）埃洛（人名）",
    },
    {
      "position": "v.",
      "translate": "说（或大声说）“喂”；打招呼",
    },
  ],
  "ukPhonetic": "həˈləʊ",
  "ukSpeech": "https://dict.youdao.com/dictvoice?hello&type=1",
  "usPhonetic": "həˈloʊ",
  "usSpeech": "https://dict.youdao.com/dictvoice?hello&type=2",
  "word": "hello",
}
`;
